src/domain/entities/Stats.ts

```diff
@@ -1,9 +1,47 @@
 import { Id } from "./Base";
 
-export type Stats = {
+type StatsAttrs = {
     recordsSkipped: Id[];
     created: number;
     ignored: number;
     updated: number;
     errorMessage: string;
 };
+
+export class Stats {
+    public readonly recordsSkipped: Id[];
+    public readonly created: number;
+    public readonly ignored: number;
+    public readonly updated: number;
+    public readonly errorMessage: string;
+
+    constructor(attrs: StatsAttrs) {
+        this.recordsSkipped = attrs.recordsSkipped;
+        this.created = attrs.created;
+        this.ignored = attrs.ignored;
+        this.updated = attrs.updated;
+        this.errorMessage = attrs.errorMessage;
```

Comment:
To avoid this duplication we have Struct in the new skeleton, but that's perfectly fine.